opentracing-examples/src/test/java/io/opentracing/examples/async_propagation/Actor.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.examples.async_propagation;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.tag.Tags;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.Phaser;
+
+/** @author tylerbenson */
+public class Actor implements AutoCloseable {
+  private final ExecutorService executor;
+  private final MockTracer tracer;
+  private final Phaser phaser;
+
+  public Actor(MockTracer tracer, Phaser phaser) {
+    // Passed along here for testing. Normally should be referenced via GlobalTracer.get().
+    this.tracer = tracer;
+
+    this.phaser = phaser;
+    executor = Executors.newSingleThreadExecutor();
```

Comment:
single thread?

Reply:
This is a simplistic example not intended as production code.  Obviously you would not want to use this class.  Only one task is ever executing on that executor, so multiple threads are unneeded.  In the promise example, there are 2 threads provisioned because they need to work concurrently.